### PR TITLE
appsettings.json=Production appsettings.Development.json=Development

### PR DESCRIPTION
**Pull request by RehanSaeed**
Fixes https://github.com/Dotnet-Boxed/Templates/issues/478.

I can't remember why I was having trouble with doing this earlier. Something could have got fixed in .NET Core that makes this work.

### CODE REVIEW

**Comment by VictorioBerra** [comment]
Looking now.

**Comment by RehanSaeed** [comment]
Good Q's. Will take a look tomorrow.

File: Source/ApiTemplate/Source/ApiTemplate/appsettings.Test.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "MinimumLevel": {
+      "Default": "Information",
+      "Override": {
+        "Microsoft": "Warning",
+        "System": "Information"
+      }
+    },
+    "WriteTo": [
+      {
+        "Name": "Console",
```
[the remark this answers is not in the document]
Comment (RehanSaeed) [reply Source/ApiTemplate/Source/ApiTemplate/appsettings.Test.json]: Yeah, I was thinking that too. Maybe only use the config for the log levels?

**Comment by RehanSaeed** [comment]
Regarding `https://localhost:5001/index.html`, your browser is caching that URL.

File: Source/ApiTemplate/Tests/ApiTemplate.IntegrationTest/Fixtures/CustomWebApplicationFactory.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ protected override void ConfigureClient(HttpClient client)
 
         protected override void ConfigureWebHost(IWebHostBuilder builder) =>
             builder
-                .UseEnvironment("Testing")
+                .UseEnvironment("Test")
```
[the remark this answers is not in the document]
Comment (RehanSaeed) [reply Source/ApiTemplate/Tests/ApiTemplate.IntegrationTest/Fixtures/CustomWebApplicationFactory.cs]: I thought the same but then tests failed. Turns out you need it for:

```
  "HostFiltering": {
    "AllowedHosts": [
      "*"
    ]
  },
```

This used to be the default but now the default is `example.com`, the tests fail. So sadly, we do need this.

File: Source/ApiTemplate/Source/ApiTemplate/appsettings.Development.json [hunk 3]
```diff
@@ -32,18 +34,15 @@
       "Default": "Information",
```
[the remark this answers is not in the document]
Comment (RehanSaeed) [reply Source/ApiTemplate/Source/ApiTemplate/appsettings.Development.json]: Switched to Debug

File: Source/ApiTemplate/Source/ApiTemplate/appsettings.Test.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (RehanSaeed) [reply Source/ApiTemplate/Source/ApiTemplate/appsettings.Test.json]: Done. See what you think.

**Comment by RehanSaeed** [comment]
@VictorioBerra How does this now look?

**Comment by VictorioBerra** [comment]
> @VictorioBerra How does this now look?

I don't think you need the environment conditional because debug does nothing in Release config, and console might always be desired. Although your way might have performance benefits. Either way is fine.

As for the `hostEnvironment.ApplicationName` that might be null in your integration tests, I haven't tested this yet but a quick Google shows an older issue: https://github.com/aspnet/Hosting/issues/1140

Like I said this might not even be an issue at all, but you may want to double check.

**Comment by RehanSaeed** [comment]
I don't recommend running console in production. It's very slow and makes a big difference to performance. If I need a conditional for console, then I might as well have debug in there as well.

Will test the `ApplicationName`.

**Comment by VictorioBerra** [comment]
> I don't recommend running console in production. It's very slow and makes a big difference to performance. If I need a conditional for console, then I might as well have debug in there as well.
> 
> Will test the `ApplicationName`.

Sounds good. LGTM then.

**Pull request merged**